src/platform/ESP32/ESP32SecureCertDACProvider.cpp

```diff
@@ -111,20 +115,61 @@ CHIP_ERROR ESP32SecureCertDACProvider ::GetProductAttestationIntermediateCert(Mu
 CHIP_ERROR ESP32SecureCertDACProvider ::SignWithDeviceAttestationKey(const ByteSpan & messageToSign,
                                                                      MutableByteSpan & outSignBuffer)
 {
+    esp_err_t esp_err;
+    esp_secure_cert_key_type_t keyType;
+
+    CHIP_ERROR chipError;
     Crypto::P256ECDSASignature signature;
-    Crypto::P256Keypair keypair;
-    char * sc_keypair       = NULL;
-    uint32_t sc_keypair_len = 0;
 
     VerifyOrReturnError(IsSpanUsable(outSignBuffer), CHIP_ERROR_INVALID_ARGUMENT);
     VerifyOrReturnError(IsSpanUsable(messageToSign), CHIP_ERROR_INVALID_ARGUMENT);
     VerifyOrReturnError(outSignBuffer.size() >= signature.Capacity(), CHIP_ERROR_BUFFER_TOO_SMALL);
 
-    esp_err_t err = esp_secure_cert_get_priv_key(&sc_keypair, &sc_keypair_len);
-    if (err == ESP_OK && sc_keypair != NULL && sc_keypair_len != 0)
+    esp_err = esp_secure_cert_get_priv_key_type(&keyType);
+    VerifyOrReturnError(esp_err == ESP_OK, CHIP_ERROR_INCORRECT_STATE,
+                        ESP_LOGE(TAG, "Failed to get the type of private key from secure cert partition, esp_err:%d", esp_err));
+
+    VerifyOrReturnError(keyType != ESP_SECURE_CERT_INVALID_KEY, CHIP_ERROR_INCORRECT_STATE,
+                        ESP_LOGE(TAG, "Private key type in secure cert partition in invalid"));
+
+    // This flow is for devices supporting ECDSA peripheral
+    if (keyType == ESP_SECURE_CERT_ECDSA_PERIPHERAL_KEY)
```

Comment:
Can this check be done during compile time instead of runtime? It helps to save some binary size.

Reply:
Key type depends on the secure cert partition being flashed, so we need to do this check here. Idea of doing this dynamic is that, irrespective of whether the ECDSA peripheral being used or not the same firmware shall work on both older(with key in partition) and newer(efuse block id of key in partition) secure cert partition.

And if you check the line just below this one, code is guarded with `SOC_ECDSA_SUPPORTED` #def so this code will only compile for H2(for now). And above check makes sure that we are using correct secure cert partition.

Reply:
I have wrapped this code around `CONFIG_USE_ESP32_ECDSA_SUPPORTED` config option now and made that option visible. Now, user can disable this option and could save some flash.